napari/components/_tests/test_viewer_model.py

```diff
@@ -760,22 +760,20 @@ def my_custom_callback():
 @pytest.mark.parametrize(
     'field', ['camera', 'cursor', 'dims', 'grid', 'layers', 'scale_bar']
 )
-def test_not_mutable_fields(field):
-    """Test appropriate fields are not mutable."""
+def test_inplace_mutable_fields(field):
+    """Test appropriate fields are mutable inplace."""
     viewer = ViewerModel()
 
     # Check attribute lives on the viewer
     assert hasattr(viewer, field)
-    # Check attribute does not have an event emitter
-    assert not hasattr(viewer.events, field)
-
-    # Check attribute is not settable
-    with pytest.raises(TypeError) as err:
-        setattr(viewer, field, 'test')
-
-    assert 'has allow_mutation set to False and cannot be assigned' in str(
-        err.value
-    )
+    # Check attribute has an event emitter
+    assert hasattr(viewer.events, field)
+
+    # Check attribute is settable, inplace
+    old = getattr(viewer, field)
+    setattr(viewer, field, old.copy())
+    new = getattr(viewer, field)
+    assert new is old
```

Comment:
could you explain this test? For me, this test will pass if silently rejected to set attribute.

Reply:
Yes, it setattr were to fail silently, old and new would be the same, but this should never happen (we should get an error from failed setattr).